src/engine/TransitivePath.cpp

```diff
@@ -318,30 +325,21 @@ std::shared_ptr<TransitivePath> TransitivePath::bindLeftOrRightSide(
   for (auto [variable, columnIndexWithType] :
        leftOrRightOp->getVariableColumns()) {
     ColumnIndex columnIndex = columnIndexWithType.columnIndex_;
-    if (columnIndex != inputCol) {
-      if (columnIndex > inputCol) {
-        columnIndexWithType.columnIndex_++;
-      } else {
-        columnIndexWithType.columnIndex_ += 2;
-      }
-      p->_variableColumns[variable] = columnIndexWithType;
-      p->_resultWidth++;
-    }
+    if (columnIndex == inputCol) { continue; }
+
+    columnIndexWithType.columnIndex_ += columnIndex > inputCol ? 1: 2;
+
+    p->_variableColumns[variable] = columnIndexWithType;
+    p->_resultWidth++;
   }
   return p;
 }
 
 // _____________________________________________________________________________
 bool TransitivePath::isBound() const {
-  return _lhs.isBound() && _rhs.isBound();
+  return _lhs.isBoundVariable() || _rhs.isBoundVariable();
```

Comment:
Add a short comment why a "constant" does not count as bound:
1. We can still bind to the variable on the other side (we have an implementation for that)
2. The constant on the "bound" side means that we don't have a variable at that side, so it is impossible to "accidentally" rebind that side.

Reply:
I think the main issue is that the concept of "being bound" isn't clearly defined. "Bound" may refer to a TransitivePath that has been bound via `bindLeftOrRightSide` function or it may refer to a TransitivePath that has a bound or a constant side.
The best solution IMO is that we define that a TransitivePath or TransitivePathSide is only bound, iff. the function `bindLeftOrRightSide` has been called. Further, we give the other concept a new name. Something like `isIdOrBound` which makes the semantics clear.
I'll do some reworks to implement this and I hope then the meaning should be apparent without comments. Let me know what you think

Reply:
I have added a docstring in TransitivePath.h to explain in a little more detail